src/lib/CrsfProtocol/CRSFParser.h

```diff
@@ -0,0 +1,35 @@
+#ifndef CRSF_PARSER_H
+#define CRSF_PARSER_H
+
+#include <targets.h>
+#include <functional>
+
+#include "crsf_protocol.h"
+#include "CRSFConnector.h"
+
+class CRSFParser {
+public:
+    bool processByte(CRSFConnector *origin, uint8_t byte, const std::function<void(const crsf_header_t *)>& foundMessage = nullptr);
+
+    // unit testing
+    void Reset()
+    {
+        telemetry_state = TELEMETRY_IDLE;
+        currentTelemetryByte = 0;
+    }
+
+private:
+    // For processing incoming bytes
+    typedef enum {
+        TELEMETRY_IDLE = 0,
+        RECEIVING_LENGTH,
+        RECEIVING_DATA
+    } telemetry_state_s;
+
+    telemetry_state_s telemetry_state = TELEMETRY_IDLE;
+    uint8_t currentTelemetryByte = 0;
+    uint8_t CRSFinBuffer[CRSF_MAX_PACKET_LEN] = {};
```

Comment:
This doesn't need to be cleared, but if you want to keep it shouldn't it use the direct initialization syntax `uint8_t CRSFinBuffer[CRSF_MAX_PACKET_LEN]{}` instead of copy initialization? I think the compiler generates the same either way in this instance, but I don't think it needs to be initialized at all.